[izruff] iP

izruff opened the pull request.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

izruff committed:
Rename main class

izruff committed:
Modify greeting message

izruff committed:
Use code blocks for multi-line string

izruff committed:
Add exit message and modify greeting

izruff committed:
Add echo feature

izruff committed:
Create main loop for receiving user input

izruff committed:
Implement add task and list tasks

izruff committed:
Add Task class

izruff committed:
Task.java: Add toString method

izruff committed:
Replace string-based tasks with Task instances

izruff committed:
Add mark and unmark feature

izruff committed:
Make Task abstract and improve code quality

izruff committed:
Add Todo, Deadline, and Event subclasses

izruff committed:
Add functionality to new task types

izruff committed:
Replace cat ASCII art

izruff committed:
Add text UI testing

izruff committed:
Add exception classes

izruff committed:
Add basic error handling

Certain errors can occur when user gives an incorrect or badly
formatted input, causing the program to exit. Let's handle some of
these possibilities, including,
* unknown command names, flags, or arguments
* missing or duplicated flags, arguments
* type validation for arguments
* out-of-range errors when accessing task list through `mark` or
  `unmark`

izruff committed:
Add delete task feature

izruff committed:
Update UI test to include delete

izruff committed:
Update UI test to include examples of error handling

izruff committed:
Add TaskList class

izruff committed:
Implement TaskList partially

izruff committed:
Implement save to file with Storage class

izruff committed:
Add data folder to .gitignore

izruff committed:
Implement Storage and save command on Mittens

izruff committed:
Implement Storage load on program start

izruff committed:
Merge branch 'branch-Level-7'

calebyyy commented on src/main/java/Mittens.java:
Good variable names. Cute cat!

calebyyy commented on src/main/java/MittensException.java:
Good variable names.  Cute cat.

calebyyy commented on src/main/java/Task.java:
Good use of boolean variable names

calebyyy commented on src/main/java/BadInputException.java:
Good variable names

calebyyy reviewed (commented):
Amazing work. Keep up the good cat drawings!!

jloh02 commented on src/main/java/Mittens.java:
```diff
@@ -0,0 +1,208 @@
+
+        while (true) {
+            System.out.print("> ");
+            String input = scanner.nextLine();
```
Would be great if you handled whitespaces after a command (e.g. "bye ") as well

jloh02 commented on src/main/java/BadInputException.java:
```diff
@@ -0,0 +1,9 @@
+public class BadInputException extends MittensException {
```
Perhaps you should have all files in a package?

jloh02 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
```
Would be great if you add a header comment for all public classes and methods

jloh02 commented on src/main/java/Mittens.java:
```diff
@@ -0,0 +1,208 @@
+
+    // Unfortunately we were unable to use the initial cat ASCII art due to the
+    // Unicode characters interfering with the UI tests.
+    private final static String GREETING_MESSAGE = """
+            
+             /\\_/\\     ____________________
+             >^,^<    / Hi, I'm Mittens!   \\
+              / \\     \\ I'm a cat! Meow :3 /
+             (___)_/   --------------------
+            """;
+
+    private final static String EXIT_MESSAGE = """
+            
+             /\\_/\\     _____________
+             >^,^<    ( Bye-bye! :3 )
+              / \\      -------------
+             (___)_/
+            """;
```
Modifiers should be given in this order `private static final` for consistency

jloh02 commented on src/main/java/BadInputException.java:
```diff
@@ -0,0 +1,9 @@
+public class BadInputException extends MittensException {
+    private static final String MITTENS_MESSAGE = "Meow?! What does that mean?";
+    private static final String HELP_MESSAGE = "Type 'help' to see a list of commands," +
```
Good naming of final variables and ensuring line length is consistent

izruff committed:
Replace String date field on Deadline class with LocalDate

izruff committed:
Replace String date fields on Event class with LocalDate

izruff committed:
Merge branch 'branch-Level-8'

izruff committed:
Add TextUi class

izruff committed:
Add abstract Ui class to inherit TextUi

izruff committed:
Add abstract Ui class to inherit TextUi

izruff committed:
Add abstract Command class

izruff committed:
Add StorageFileException class and handle errors in save and load methods

izruff committed:
Add InitializationException and handle possible errors

izruff committed:
Fix typo in showCommandCompletion

izruff committed:
Refactor Mittens to use TaskList methods

izruff committed:
Rewrite Ui class and implement methods in TextUi

izruff committed:
Add command subclass for adding a new task

izruff committed:
Partially implement Command and refactor Mittens

The Mittens class violates the SLAP principle. The Command class, along
with TaskList, Ui, and Storage aims to fix this.

This commit serves as confirmation that the prototype AddCommand class
works as expected.

izruff committed:
Make small changes for style consistency

izruff committed:
Overload showRegularMessage method for convenience

izruff committed:
Add command subclass for deleting a task

izruff committed:
Modify Command and its subclasses

Several fixes and nits are included here:
* remove redundant fields result and error
* remove type parameter from Command class
* add missing call to super's constructor\
* remove unused imports

izruff committed:
Add command subclass for listing all tasks

izruff committed:
Overload showRegularMessage with varargs

izruff committed:
Add command subclasses for marking and unmarking a task

izruff committed:
Add command subclass for exiting the program

izruff committed:
Replace class methods in Mittens with use of Command instances

izruff committed:
Add CommandParser class

izruff committed:
Implement CommandParser in Mittens

izruff committed:
Refactor Mittens to utilize all other classes

izruff committed:
Merge branch 'branch-A-MoreOOP'

izruff committed:
Move all files to the mittens package

izruff committed:
Move UI-related files to mittens.ui

izruff committed:
Add getter methods to Task and its subclasses

izruff committed:
Use getter methods instead of direct access to fields

izruff committed:
Move task-related classes to mittens.task

izruff committed:
Move storage-related classes to mittens.storage

izruff committed:
Move commands to mittens.commands

izruff committed:
Move parser-related classes to mittens.parser

izruff committed:
Merge branch 'branch-A-Packages'

izruff committed:
Merge branch 'add-gradle-support'

izruff committed:
Update name to Mittens

izruff committed:
Add checkstyle plugin to Gradle

izruff committed:
Write tests for CommandParser

izruff committed:
Write tests for Storage

izruff committed:
Merge branch 'branch-A-JUnit'

izruff committed:
fix: Index off by 1 on delete, mark, and unmark commands

izruff committed:
fix: Remove extra newline from certain messages

izruff committed:
Add JavaDoc comments to Command and its subclasses

izruff committed:
Add JavaDoc comments to Mittens class

izruff committed:
Add JavaDoc comments to CommandParser and related classes

izruff committed:
Add JavaDoc comments to exception classes

izruff committed:
Add JavaDoc comments to Storage and related classes

izruff committed:
Add JavaDoc comments to Task and TaskList

izruff committed:
Add JavaDoc comments to Ui and its subclasses

izruff committed:
Enforce explicit import statements

izruff committed:
Auto-format code with Intellij code style

izruff committed:
Add find tasks by keyword functionality to TaskList

izruff committed:
Add FindCommand for finding tasks by given keyword

izruff committed:
Restore deleted part of message

izruff committed:
Fix typo

izruff committed:
Update CommandParser to detect find commands

izruff committed:
Merge branch 'branch-A-JavaDoc'

izruff committed:
Merge branch 'branch-A-CodingStandard'

izruff committed:
Merge branch 'branch-Level-9'

izruff committed:
Fix missing import statement

izruff committed:
Mittens.java: Refactor code and add method to load task list after initialization

izruff committed:
Update dependencies

izruff committed:
Implement JavaFxUi and refactor Ui class

izruff committed:
Implement JavaFX GUI application

izruff committed:
Remove redundant object

izruff committed:
Fix incorrect error message format

izruff committed:
Use PipedInputStream for user input processing

izruff committed:
Merge branch 'branch-Level-10'

izruff committed:
Storage.java: Assert valid tasks types on save

izruff committed:
Mittens.java: Assert loaded task list not null

izruff committed:
Add RawCommandElements to represent command string

izruff committed:
Implement RawCommandElements in commands and parser

izruff committed:
Merge pull request #2 from izruff/branch-A-Assertions

Add assertions

izruff committed:
Merge branch 'master' into branch-A-CodeQuality

izruff committed:
Merge pull request #3 from izruff/branch-A-CodeQuality

Improve CommandParser code quality

izruff committed:
Fix unhandled out-of-bounds error

izruff committed:
Remove redundant constructors

izruff committed:
Remove unused imports

izruff committed:
Remove unused method

izruff committed:
Update Mittens error message

izruff committed:
Modify image size to fit more messages

izruff committed:
Add UI screenshot for docs

izruff committed:
docs/README.md: Add product website

izruff committed:
Update product website

izruff committed:
Fix Ui.png not showing on website
